docs/static/playground/playground.js

```diff
@@ -1,404 +1,266 @@
-
-            var currentModel = editor.getModel();
-            if (currentModel === data.js.model) {
-                data.js.state = currentState;
+        ]
+    }, {
+        name: "Micro:bit",
+        id: "microbit",
+        endpoints: [
+            {
+                name: "beta",
+                url: "https://makecode.microbit.org/beta?controller=1"
+            },
+            {
+                name: "released",
+                url: "https://makecode.microbit.org?controller=1"
             }
-
-            editor.setModel(data[desiredModelId].model);
-            editor.restoreViewState(data[desiredModelId].state);
-            editor.focus();
-        }
-
-
-        // create the typing side
-        var typingContainer = document.createElement('div');
-        typingContainer.className = 'typingContainer';
-
-        var tabArea = (function () {
-            var tabArea = document.createElement('div');
-            tabArea.className = 'tabArea';
-
-            var jsTab = document.createElement('span');
-            jsTab.className = 'tab active';
-            jsTab.appendChild(document.createTextNode('TypeScript'));
-            jsTab.onclick = function () { changeTab(jsTab, 'js'); };
-            tabArea.appendChild(jsTab);
-
-            var runBtn = document.createElement('span');
-            runBtn.className = 'action run';
-            runBtn.appendChild(document.createTextNode('Run'));
-            runBtn.onclick = function () { run(); };
-            tabArea.appendChild(runBtn);
-
-            return tabArea;
-        })();
-
-        var editorContainer = document.createElement('div');
-        editorContainer.className = 'editor-container';
-
-        typingContainer.appendChild(tabArea);
-        typingContainer.appendChild(editorContainer);
-
-        var runContainer = document.createElement('div');
-        runContainer.className = 'run-container';
-
-        var sampleSwitcher = document.createElement('select');
-        var sampleChapter;
-        PLAY_SAMPLES.forEach(function (sample) {
-            if (!sampleChapter || sampleChapter.label !== sample.chapter) {
-                sampleChapter = document.createElement('optgroup');
-                sampleChapter.label = sample.chapter;
-                sampleSwitcher.appendChild(sampleChapter);
+        ]
+    }, {
+        name: "LEGO EV3",
+        id: "ev3",
+        endpoints: [
+            {
+                name: "beta",
+                url: "https://makecode.mindstorms.com/beta?controller=1"
+            },
+            {
+                name: "released",
+                url: "https://makecode.mindstorms.com?controller=1"
             }
-            var sampleOption = document.createElement('option');
-            sampleOption.value = sample.id;
-            sampleOption.appendChild(document.createTextNode(sample.name));
-            sampleChapter.appendChild(sampleOption);
+        ]
+    }
+];
+var CUSTOM_FILE = "_custom.ts";
```

Comment:
why not just "custom.ts" ? (not that it matters at all)

Reply:
oh i use the underscore for files that users shouldn't modify, i feel like it makes the file look less... friendly? just habit in this case haha

Reply:
the editor creates a "custom.ts" file so let's drop the ``_``